sdk/python/feast/infra/redis_provider.py

```diff
@@ -0,0 +1,281 @@
+import json
+import struct
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import mmh3
+import pandas as pd
+from google.protobuf.timestamp_pb2 import Timestamp
+from redis import Redis
+from rediscluster import RedisCluster
+from tqdm import tqdm
+
+from feast import FeatureTable, utils
+from feast.entity import Entity
+from feast.feature_view import FeatureView
+from feast.infra.offline_stores.helpers import get_offline_store_from_sources
+from feast.infra.provider import (
+    Provider,
+    RetrievalJob,
+    _convert_arrow_to_proto,
+    _get_column_names,
+    _run_field_mapping,
+)
+from feast.protos.feast.storage.Redis_pb2 import RedisKeyV2 as RedisKeyProto
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import RedisOnlineStoreConfig, RedisType, RepoConfig
+
+EX_SECONDS = 253402300799
+
+
+class RedisProvider(Provider):
+    _redis_type: Optional[RedisType]
+    _redis_connection_string: str
+
+    def __init__(self, config: RepoConfig):
+        assert isinstance(config.online_store, RedisOnlineStoreConfig)
+        if config and config.online_store:
```

Comment:
It's not clear why there is a conditional here if `isinstance` implies that this config exists. I think you should assume that `config.online_store` always exists, and we should provide defaults to these options in all cases. So you should just reach for the config without testing whether they exist.